src/ambianic/pipeline/store.py

```diff
@@ -108,7 +108,10 @@ def _save_sample(self,
             json.dump(save_json, f, ensure_ascii=False, indent=4,
                       cls=JsonEncoder)
         # e = PipelineEvent('Detected Objects', type='ObjectDetection')
-        self.event_log.info('Detection Event', save_json)
+
+        json_encoded_obj = json.loads(json.dumps(save_json, cls=JsonEncoder))
```

Comment:
@bhavikapanara Why serialize to json twice within 5 lines of code. Its an obvious redundancy. Serialize once to a json string and reuse it: Once for standalone file write and then for deserialization before passing to event_log.

```
json_str = json.dumps(save_json, cls=JsonEncoder)
with open(json_path, 'w', encoding='utf-8') as f:
  f.write(json_str)
json_encoded_obj = json.loads(json_str)
```

Reply:
@ivelin yes I know It is redundancy.

But there was an issue with argument in `event_log.info()`. Let me explain it here:

if I use the below line: 

```
json_enc = json.dumps(save_json, cls=JsonEncoder)
self.event_log.info('Detection Event', json_enc)
```

It throws the error:

```
File "/workspace/src/ambianic/pipeline/store.py", line 114, in _save_sample
    self.event_log.info('Detection Event', json_enc)
Message: 'Detection Event'
Arguments: ('{"id": "5c2bcf6031164129a65ec031df55f416", "datetime": "2021-03-01T17:55:48.232017", "image_file_name": "20210301-175548.232017-image.jpg", "thumbnail_file_name": "20210301-175548.232017-thumbnail.jpg", "json_file_name": "20210301-175548.232017-inference.json", "rel_dir": "detections/20210301-175547.531457", "inference_result": [{"label": "person", "confidence": 0.9921875, "box": {"xmin": 0.3868940770626068, "ymin": 0.12535724414170846, "xmax": 0.7228575944900513, "ymax": 1}}], "inference_meta": {"display": "Object Detection"}}',)
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.7/logging/handlers.py", line 68, in emit
    if self.shouldRollover(record):
  File "/usr/lib/python3.7/logging/handlers.py", line 184, in shouldRollover
    msg = "%s\n" % self.format(record)
  File "/usr/lib/python3.7/logging/__init__.py", line 880, in format
    return fmt.format(record)
  File "/usr/lib/python3.7/logging/__init__.py", line 619, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.7/logging/__init__.py", line 380, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting

```

As it won't allow adding string argument in self.event_log.info('Detection Event', json_enc)

I have also tried below solution:

```
json_enc = json.dumps(save_json, cls=JsonEncoder)
self.event_log.info('Detection Event=%s', json_enc)

```

This was able to fix the `TypeError: not all arguments converted during string formatting` but it broke the UI with the below error.

![image](https://user-images.githubusercontent.com/19815572/109540389-5f9c1d00-7ae8-11eb-9858-24e9cc7f5563.png)


Then I decided to decode the JSON string again and convert it to a list object and pass it to `self.event_log.info()`

```
json_encoded_obj = json.loads(json.dumps(save_json, cls=JsonEncoder))
self.event_log.info('Detection Event', json_encoded_obj)
```

Here the [json.dump()](https://github.com/bhavikapanara/ambianic-edge/blob/7026529e055c75b44bb990c9b54c6128f660a841/src/ambianic/pipeline/store.py#L108) is used to write data in JSON file and it won't return anything.

However, json.dumps() return encoded string. 
So, to remove the redundancy we can use this json encoded string to save json file. 

But there was an issue with an indent in a saved json file.

Still, I am searching a better solution to remove redundancy. Please let me know your thoughts.

Reply:
@bhavikapanara look closer at my example code. You need to still deserialize from the encoded json string and then pass that python object to `event_log`. You missed this part:

```
json_encoded_obj = json.loads(json_str)
```